handbook/engineering/scheduled-maintenance.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Scheduled maintenance
+titleTemplate: :title | Engineering | Tuist Handbook
+description: This document outlines our company's approach to communicating scheduled maintenance activities to users.
+---
+
+## Scheduled Maintenance Guidelines
+
+### Overview
+
+This document outlines our company's approach to communicating scheduled maintenance activities to users.
+Following these guidelines ensures transparent and timely communication about system maintenance and potential service interruptions.
+
+| Category | Expected Impact                    | First Notice |
+| -------- | ---------------------------------- | ------------ |
+| Minor    | No expected downtime, minimal risk | 24 hours     |
+| Standard | Limited service interruption       | 3 days       |
+| Major    | Significant service interruption   | 7 days       |
+
+### Messaging
+
+All maintenance announcements, regardless of category, should include:
+
+- Start date and time (including timezone)
+- Expected duration
+- Components affected
+- Expected user impact
+- Link to status page for updates
+
+### Maintenance Categories
+
+The following categories should be considered as guidelines for scheduling maintenance activities. Depending on the activities and
+associated risks, the category may be adjusted accordingly.
+
+#### Minor Maintenance
+
+**Expected Impact**: No expected downtime, minimal risk
+
+Minor maintenance is a low-risk maintenance activity that we do not expect to cause a service interruption. We communicate these activities
+for transparency and for risk management.
+
+##### Communication Timeline
+
+- First notice: 24 hours before
+- Reminder: 1 hour before
+
+##### Communication Channels
+
+- Status page
+- Slack announcement
+
+#### Standard Maintenance
+
+**Expected Impact**: Limited service interruption  
+**Duration**: 5-60 minutes
+
+Standard maintenance is an activity that we expect to cause a limited service interruption, meaning that at least part of the service
+is unavailable for a short period of time.
+
+##### Communication Timeline
+
+- First notice: 3 days before
+- Reminders:
+  - 24 hours before
+  - 1 hour before
+
+##### Communication Channels
+
+- Status page
+- Slack announcement
+- Community forum post
+
+#### Major Maintenance
+
+**Expected Impact**: Significant service interruption  
+**Duration**: 1 hour or longer
+
+A major maintenance activity is an activity that we expect to cause a significant service interruption, meaning that part or the entirety of
+the service is unavailable for an extended period of time.
+
+##### Communication Timeline
+
+- First notice: 7 days before
+- Reminders:
+  - 3 days before
+  - 24 hours before
+  - 1 hour before
+
+##### Communication Channels
+
+- Status page
+- Slack announcement
+- Social media announcement
+- Community forum post
+- Direct message to enterprise users
```

Comment:
Should we instead message all users via email? Such a notice could also be done for minor maintenance events, but given the limited impact, I wouldn't say it's necessary.

Reply:
I agreed. We'll also need to figure out a system for doing this.